frontend/src/pages/detail/GenreDetail.tsx

```diff
@@ -1,85 +1,72 @@
-import React, { FormEvent } from "react";
+import React, { FormEvent, useState } from "react";
 import { InputText } from "primereact/inputtext";
 import { ToggleButton } from "primereact/togglebutton";
 import ConfirmButton from "../../components/ConfirmButton";
+import { useLocation } from "react-router-dom";
 
-interface modifyGenreState {
+export interface GenreDetailState {
   genre: string;
   isModifiable: boolean;
-  isConfirmationPopVisible: boolean;
+  isConfirmationPopupVisible: boolean;
 }
 
-class ModifyGenrePage extends React.Component<{}, modifyGenreState> {
-  constructor(props = {}) {
-    super(props);
-    this.state = {
-      genre: "asdfasv",
-      isModifiable: false,
-      isConfirmationPopVisible: false,
-    };
-  }
+export default function GenreDetail() {
+  const location = useLocation();
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const detailState = location.state! as GenreDetailState;
```

Comment:
Just a note because I worked around this with fact const data. Visiting the detail pages currently through errors and displays blank screen. Its because of no data value being fed into the state functions at the top. Just wanted to acknowledge that.

Reply:
Yeah I fixed this on my new branch that links to the detail page, so no worries on this. Thanks for pointing it out